File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> hash;
        for (int i = 0; i<nums.size(); i++) {
            int cmp = target - nums[i];
            if (hash.find(cmp) != hash.end()) {
                return {i, hash[cmp]};
            }

            hash[nums[i]] = i;
        }
        return {};
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> map = new HashMap<>();
        int n = nums.length;
        for (int i = 0; i < n; i++) {
            map.put(nums[i], i);
        }

        for (int i = 0; i < n; i++) {
            int cmp = target - nums[i];
            if (map.containsKey(cmp) && map.get(cmp) != i) {
                return new int[]{i, map.get(cmp)};
            }
        }

        return new int[]{};
    }
}